third_party/terraform/tests/resource_compute_router_nat_test.go

```diff
@@ -0,0 +1,305 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccComputeRouterNat_basic(t *testing.T) {
+	t.Parallel()
+
+	testId := acctest.RandString(10)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeRouterNatDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeRouterNatBasic(testId),
+				Check: testAccCheckComputeRouterNatExists(
```

Comment:
Since you have the import step, you don't need a separate check here (since if the nat doesn't exist, the import will fail)

Reply:
Right, the import also does that- importing a resource reads it from the API and puts its values in state. The `ImportStateVerify` line tells the test infrastructure to check that the state when the resource was created matches the state when the resource is imported.

Reply:
Got it. Done.